File: app/src/main/java/net/tlalka/android/fiszki/models/dto/LessonDto.java
package net.tlalka.android.fiszki.models.dto;

import net.tlalka.android.fiszki.models.entities.Lesson;
import net.tlalka.android.fiszki.models.types.LevelType;
import org.parceler.Parcel;

@Parcel
public class LessonDto {

    long lessonId;
    String lessonName;
    LevelType lessonLevel;

    public LessonDto() {
    }

    public LessonDto(Lesson lesson) {
        this.lessonId = lesson.getId();
        this.lessonName = lesson.getName();
        this.lessonLevel = lesson.getLevelType();
    }

    public long getLessonId() {
        return lessonId;
    }

    public String getLessonName() {
        return lessonName;
    }

    public LevelType getLessonLevel() {
        return lessonLevel;
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/model/dto/parcel/LessonDto.kt
package net.tlalka.fiszki.model.dto.parcel

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import net.tlalka.fiszki.model.entities.Lesson
import net.tlalka.fiszki.model.types.LevelType

@Parcelize
data class LessonDto constructor(val lessonId: Long = 0,
                                 val lessonName: String = "",
                                 val lessonLevel: LevelType = LevelType.BEGINNER,
                                 val lessonIndex: Int = 0,
                                 var generalScore: Int = 0,
                                 var correctScore: Int = 0,
                                 var incorrectScore: Int = 0) : Parcelable {

    constructor(lesson: Lesson, index: Int) : this(lesson.id, lesson.name, lesson.levelType, index)

    fun setScoreValues(generalScore: Int, correctScore: Int, incorrectScore: Int) {
        this.generalScore = generalScore
        this.correctScore = correctScore
        this.incorrectScore = incorrectScore
    }
}
